### PR TITLE
Remind we support alter table with generated columns except for drop …

### LINKED ISSUE #2379: Document: feat(ddl): allow alter table with generated columns
This issue tracks the documentation update needed for the merged PR #17652.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/17652
Source PR Merged At: 2024-07-15T16:21:08Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
…columns

## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/17652

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2379

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/sql/commands/sql-alter-table.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ slug: /sql-alter-table
 
 The `ALTER TABLE` command modifies the definition of a table.
 
+:::
+For tables with generated columns, we support altering them with the clauses below, except for `DROP COLUMN`. If you attempt to drop a column that's referenced by a generated column, RisingWave will return an error. To resolve this, you can first drop the generated column, and then drop the referenced column.
+:::
+
```
Comment (ShanlanLi) [inline docs/sql/commands/sql-alter-table.md]: @yuhao-su Just want to double-check. We now support all the clauses on this page except for the `drop column`. Is it right?

Comment (yuhao-su) [reply docs/sql/commands/sql-alter-table.md]: > To resolve this, you can first drop the generated column, and then drop the referenced column.

We do not support drop any column now.

> we support altering them with the clauses below, except for `DROP COLUMN`

We only support refresh schema.

Comment (ShanlanLi) [reply docs/sql/commands/sql-alter-table.md]: > We do not support drop any column now.

Thanks, but the code pr said "User can proceed drop after dropping the generated column"
image

And I just searched in the doc, finding no place saying we don't support altering tables with generated columns. So there is no need to delete any note. Then how about just adding the following note below the `DROP COLUMN` command:

```
If you attempt to drop a column referenced by a generated column, RisingWave will return an error. To resolve this, you can first drop the generated column, and then drop the referenced column.
```

Comment (yuhao-su) [reply docs/sql/commands/sql-alter-table.md]: > but the code pr said "User can proceed drop after dropping the generated column"

There is a TODO below this. Finished and doced in #2405

> Then how about just adding the following note below the DROP COLUMN command

Sure we can do that

File: docs/sql/commands/sql-alter-table.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ ALTER TABLE employees DROP fax;
 
 + You cannot drop columns referenced by materialized views or indexes.
 
-+ If you attempt to drop a column referenced by a generated column, RisingWave will return an error. To resolve this, you can first drop the generated column, and then drop the referenced column.
++ You cannot drop columns referenced by a generated column without the generated column being dropped.
```
Comment (yuhao-su) [inline docs/sql/commands/sql-alter-table.md]: > being dropped

Should this be _been dropped_?

Comment (WanYixian) [reply docs/sql/commands/sql-alter-table.md]: How about "To drop a column referenced by a generated column, you must first drop the generated column."

Comment (yuhao-su) [reply docs/sql/commands/sql-alter-table.md]: SGTM

**Review by WanYixian — APPROVED** [review]

**Pull request merged**